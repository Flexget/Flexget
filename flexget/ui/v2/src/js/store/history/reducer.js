import { GET_HISTORY } from 'store/history/actions';

const initState = {
  totalCount: 0,
  items: [],
};


export default function reducer(state = initState, { type, payload }) {
  switch (type) {
    case GET_HISTORY:
      return {
        totalCount: payload.headers.get('total-count'),
        items: payload.refresh ? payload.data : [...state.items, ...payload.data],
      };
    default:
      return state;
  }
}
