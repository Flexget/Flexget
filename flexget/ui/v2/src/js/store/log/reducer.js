import {
  LOG_CONNECT,
  LOG_MESSAGE,
  LOG_DISCONNECT,
  LOG_CLEAR,
} from 'store/log/actions';

const initState = {
  messages: [],
  connected: false,
};

export default function reducer(state = initState, action) {
  switch (action.type) {
    case LOG_CONNECT:
      return {
        ...state,
        messages: [],
        connected: true,
      };
    case LOG_MESSAGE:
      return {
        ...state,
        messages: [action.payload, ...state.messages],
      };
    case LOG_DISCONNECT:
      return {
        ...state,
        connected: false,
      };
    case LOG_CLEAR:
      return {
        ...state,
        messages: [],
      };
    default:
      return state;
  }
}
