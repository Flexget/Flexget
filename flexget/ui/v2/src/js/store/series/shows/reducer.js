import { GET_SHOWS } from 'store/series/shows/actions';

const initState = {
  totalCount: 0,
  items: [],
};

export default function reducer(state = initState, { type, payload }) {
  switch (type) {
    case GET_SHOWS:
      return {
        totalCount: payload.headers.get('total-count'),
        items: payload.data,
      };
    default:
      return state;
  }
}
