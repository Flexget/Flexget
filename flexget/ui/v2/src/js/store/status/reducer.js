import {
  LOADING_STATUS,
  ERROR_STATUS,
  CLOSE_STATUS,
  INFO_STATUS,
} from 'store/status/actions';
import { LOCATION_CHANGE } from 'connected-react-router';

const initState = {
  loading: {},
  info: null,
  error: null,
};

export default (state = initState, action) => {
  switch (action.type) {
    case LOADING_STATUS:
      return {
        ...state,
        loading: {
          ...state.loading,
          [action.meta.type]: true,
        },
      };
    case ERROR_STATUS: {
      const {
        [action.payload.type]: omit,
        ...loading
      } = state.loading;
      return {
        ...state,
        error: action.payload,
        loading,
      };
    }
    case INFO_STATUS:
      return {
        ...state,
        info: action.payload.message,
      };
    case CLOSE_STATUS:
    case LOCATION_CHANGE:
      return {
        ...initState,
        loading: state.loading,
      };
    default:
      if (state.loading[action.type]) {
        const {
          [action.type]: omit,
          ...loading
        } = state.loading;
        return {
          ...state,
          loading,
        };
      }
      return state;
  }
};
