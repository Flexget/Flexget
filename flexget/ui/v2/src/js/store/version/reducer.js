import { GET_VERSION } from 'store/version/actions';

export default function reducer(state = {}, action) {
  switch (action.type) {
    case GET_VERSION: {
      const {
        api_version: api,
        flexget_version: flexget,
        latest_version: latest,
      } = action.payload;

      return { api, flexget, latest };
    }
    default:
      return state;
  }
}
