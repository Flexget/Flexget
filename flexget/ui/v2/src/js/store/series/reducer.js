import { combineReducers } from 'redux';
import shows from 'store/series/shows/reducer';

export default combineReducers({
  shows,
});
