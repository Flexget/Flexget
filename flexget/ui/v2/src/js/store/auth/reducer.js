import { LOGIN, LOGOUT } from 'store/auth/actions';
import { ERROR_STATUS } from 'store/status/actions';
import { GET_VERSION } from 'store/version/actions';

export default function reducer(state = {}, action) {
  switch (action.type) {
    case LOGIN:
    case GET_VERSION:
      return {
        loggedIn: true,
      };
    case LOGOUT:
      return {};
    case ERROR_STATUS:
      if (action.payload.statusCode === 401) {
        return {};
      }
    default: // eslint-disable-line no-fallthrough
      return state;
  }
}
