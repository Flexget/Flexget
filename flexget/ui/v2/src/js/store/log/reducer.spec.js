import reducer from 'store/log/reducer';
import {
  LOG_CONNECT,
  LOG_MESSAGE,
  LOG_DISCONNECT,
  LOG_CLEAR,
} from 'store/log/actions';

describe('store/log/reducer', () => {
  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should be connected on LOG_CONNECT', () => {
    expect(reducer(undefined, { type: LOG_CONNECT }).connected).toBe(true);
  });

  it('should add the messages to the existing messages', () => {
    expect(reducer(undefined, { type: LOG_MESSAGE, payload: ['some messages'] })).toMatchSnapshot();
  });

  it('should be disconnected on LOG_DISCONNECT', () => {
    expect(reducer({ connected: true }, { type: LOG_DISCONNECT }).connected).toBe(false);
  });

  it('should set lines on LOG_CLEAR', () => {
    expect(reducer({ messages: [{ not: 'empty' }] }, { type: LOG_CLEAR }).messages).toHaveLength(0);
  });
});
