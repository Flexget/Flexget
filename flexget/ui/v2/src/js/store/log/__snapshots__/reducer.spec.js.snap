// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store/log/reducer should add the messages to the existing messages 1`] = `
Object {
  "connected": false,
  "messages": Array [
    Array [
      "some messages",
    ],
  ],
}
`;

exports[`store/log/reducer should return the initial state 1`] = `
Object {
  "connected": false,
  "messages": Array [],
}
`;
